include/targets/cuda/reduce_helper.h

```diff
@@ -209,6 +220,8 @@ namespace quda
             sum_tmp[i] = sum_tmp[i] == init_value<atomic_t>() ? terminate_value<atomic_t>() : sum_tmp[i];
             arg.result_d[n * idx + i].store(sum_tmp[i], cuda::std::memory_order_relaxed);
           }
+        } else if (arg.device_output_buffer) {
+          arg.get_device_output_buffer()[idx] = sum;
```

Comment:
The implication here is that if the manually set device buffer is used, there is no synchronization possible since it doesn't use an atomic write.  With this in mind, I think the getter and setter functions should labelled `device_output_async_buffer` to make this more explicit, and note this in the doxygen comments for these functions.

Reply:
And thinking about this: I'm not sure we need the name `device` in the title.  It could be a pinned host buffer for example, or if this function was run on the host (in some future code evolution), then the `device` name would be misleading.  So how about `get_output_async_buffer` and `set_output_async_buffer`?

Reply:
But then we need a way to tell if it is a device buffer or host buffer right?

Reply:
Yes, but that logic would be handled at a higher level when one is deciding to run on host or device.  For now, since this buffer is always on the device, there is no risk of ambiguity.  And it's just a name 😄

Reply:
Sound good. I will change to `get_output_async_buffer` and `set_output_async_buffer`.

Reply:
Discussed offline and I should check if the async knob is turned on when the setter function is called.

Reply:
https://github.com/lattice/quda/pull/1061/commits/d9e668e4f01f639c0fb01188d4fbbcd2cbf34544